mycore-base/src/main/java/org/mycore/common/MCRUserInformation.java

```diff
@@ -53,4 +53,10 @@
      */
     String getUserAttribute(String attribute);
 
+    /**
+     * Get all attribute names if they are provided by the underlying user system
+     * @return user attribute names as String array or null if no value is defined
+     */
+    String[] getUserAttributeNames(); //TODO maybe more generic
+
```

Comment:
Do not extend the interface. We do not ask for a list of attributes but ask for a specific attribute.

Reply:
Which attributes there are depends on the application. I don't know that explicitly at this point, that's why I chose the approach. Alternatively, prefixes for attributes could be explicitly defined in the configuration and queried.